cmd/permissions-checker/permissionschecks.go

```diff
@@ -0,0 +1,117 @@
+			currentPerm := mode.Perm().String()
+			hasCorrectOwner := checkOwner(currentOwner.Username, DefaultOwner)
+			hasCorrectGroup := checkOwner(currentGroup.Name, DefaultGroup)
+
+			if mode.IsRegular() && (f.Name()[0:1] != "." || ShowHidden) {
+				hasCorrectPermissions := checkPermissions(currentPerm, DefaultFilePerm)
+				printOutput(level, "f", name, fullPath, currentPerm, DefaultFilePerm, currentOwner.Username, currentGroup.Name, hasCorrectPermissions, hasCorrectOwner, hasCorrectGroup)
+			} else if mode.IsDir() && (f.Name()[0:1] != "." || ShowHidden) {
+				hasCorrectPermissions := checkPermissions(currentPerm, DefaultDirPerm)
+				printOutput(level, "d", name, fullPath, currentPerm, DefaultDirPerm, currentOwner.Username, currentGroup.Name, hasCorrectPermissions, hasCorrectOwner, hasCorrectGroup)
+				FindRecursive(fullPath, level)
+			}
+		}
+	}
+}
+
+// checkPermissions return true if the permissions are correct (false in other case)
+/* currentPermissions [string] - permissions that the file or directory has
+ * defaultPermissions [string] - permissions that should have
+ */
+func checkPermissions(currentPermissions, defaultPermissions string) bool {
+	return strings.Contains(currentPermissions, defaultPermissions)
+}
+
+// checkOwner return true if the owner is correct (false in other case)
+/* currentOwner [string] - owner of the file or directory
+ * defaultOwner [string] - owner that should have the file or directory
+ */
+func checkOwner(currentOwner, defaultOwner string) bool {
+	if currentOwner == defaultOwner {
+		return true
+	}
+	return false
+}
+
+// checkGroup return true if the group is correct (false in other case)
+/* currentGroup [string] - group of the file or directory
+ * defaultGroup [string] - group that should have the file or directory
+ */
+func checkGroup(currentGroup, defaultGroup string) bool {
+	if currentGroup == defaultGroup {
+		return true
+	}
+	return false
+}
+
+// printOutput Print the data in different formats according to the situation
+/* level [string] 							- used to print the output in a hierarchical way
+ * kind [string] 								- "f" if it is file or "d" in case of a directory
+ * name [string] 								- file/directory name
+ * fullPath [string] 						- full path of the file/directory
+ * currentPermissions [string] 	- file/directory current permissions
+ * defaultPermissions [string] 	- file/directory default permissions
+ * currentOwner [string] 				- file/directory current owner
+ * currentGroup [string] 				- file/directory current group
+ * hasCorrectPermissions [bool] - true if the permissions are correct
+ * hasCorrectOwner [bool] 			- true if the owner is correct
+ * hasCorrectGroup [bool] 			- true if the group is correct
+ */
+func printOutput(level, kind, name, fullPath, currentPermissions, defaultPermissions, currentOwner, currentGroup string, hasCorrectPermissions, hasCorrectOwner, hasCorrectGroup bool) {
+	if !Verbose {
```

Comment:
I would remove the need of providing both `name` and `fullPath` by using `filepath.Base`:
```name := fullPath
if !Verbose {
    name = filepath.Base(fullPath)
}
```
Note the logic is inverted to avoid calculating the base name unnecessarily